visualization/app/codeCharta/state/store/metricData/edgeMetricData/edgeMetricData.actions.ts

```diff
@@ -0,0 +1,36 @@
+import { Action } from "redux"
+import { BlacklistItem, EdgeMetricData } from "../../../../codeCharta.model"
+import { FileState } from "../../../../model/files/files"
+
+export enum EdgeMetricDataActions {
+	SET_EDGE_METRIC_DATA = "SET_EDGE_METRIC_DATA",
+	CALCULATE_NEW_EDGE_METRIC_DATA = "CALCULATE_NEW_EDGE_METRIC_DATA"
+}
+
+export interface SetEdgeMetricDataAction extends Action {
+	type: EdgeMetricDataActions.SET_EDGE_METRIC_DATA
+	payload: EdgeMetricData[]
+}
+
+export interface CalculateNewEdgeMetricDataAction extends Action {
+	type: EdgeMetricDataActions.CALCULATE_NEW_EDGE_METRIC_DATA
+	payload: { fileStates: FileState[]; blacklist: BlacklistItem[] }
+}
+
+export type EdgeMetricDataAction = SetEdgeMetricDataAction | CalculateNewEdgeMetricDataAction
+
+export function setEdgeMetricData(edgeMetricData: EdgeMetricData[] = defaultEdgeMetricData): SetEdgeMetricDataAction {
```

Comment:
The name `set...` confuses me. This function is only going to return a new object but it is not going to set anything?

Since the function returns the passed in edgeMetricData 1-to-1, we might also just remove this function and just create the object where it's required

Reply:
Those functions are called `ActionCreators`. These function return an action, which is an object with a type and an optional payload. You put these ActionCreators in the dispatch function, e.g. `storeService.dispatch(setEdgeMetricData([]))`
Don't get to confused by the function itself. It makes much more sense when using the function with dispatch.

Reply:
I see. Is this a common pattern for Redux?

Reply:
Yes, check out actions [here](https://redux.js.org/basics/actions)
There are some patterns to [reduce this boilerplate code](https://redux.js.org/recipes/reducing-boilerplate). Applying that would solve your confusion, I guess. :)

Reply:
TIL. Thanks for the link!